c7n/resources/cw.py

```diff
@@ -21,12 +21,36 @@
 from c7n.resolver import ValuesFrom
 from c7n.resources import load_resources
 from c7n.resources.aws import ArnResolver
+from c7n.query import RetryPageIterator
 from c7n.tags import universal_augment
 from c7n.utils import type_schema, local_session, chunks, get_retry, jmespath_search
 from botocore.config import Config
 import re
 
 
+# https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/events.html#paginators
+# There is no paginator for list_event_buses. Therefore manual pagination is implemented here.
+def GetEventBuses(client):
+    event_buses = []
```

Comment:
paginators are a base class that can be used directly with pagination, we have several examples extant in custodian
```
 grep Paginator *    
grep: __pycache__: Is a directory
awslambda.py:from botocore.paginate import Paginator
awslambda.py:    pager = Paginator(
route53.py:from botocore.paginate import Paginator
route53.py:    return Paginator(
shield.py:from botocore.paginate import Paginator
shield.py:    return Paginator(
```

which would be preferred for consistency and code reduction

Reply:
Code updated. Thanks!